src/Cafe/HW/Latte/LegacyShaderDecompiler/LatteDecompilerAnalyzer.cpp

```diff
@@ -850,9 +850,12 @@ void LatteDecompiler_analyze(LatteDecompilerShaderContext* shaderContext, LatteD
 			cemu_assert_unimplemented();
 	}
 	// analyze subroutines
-	for (auto subroutineAddr : list_subroutineAddrs)
+	if (!list_subroutineAddrs.empty())
```

Comment:
What is the advantage of checking if non-empty for a range based loop?

Reply:
Yes, you're right, after C++11, don't need write this
Reference: https://stackoverflow.com/questions/52828423/in-c11-and-later-is-it-still-optimal-to-do-an-empty-check-before-a-range-f